src/app/routes/auth.js

```diff
@@ -205,92 +205,52 @@ export default new Hono()
   })
   .post("/login/magic-link", async (c) => {
     const body = await c.req.parseBody();
-    const { email } = body;
-
-    try {
-      console.log("Calling signInMagicLink with:", { email });
-      const result = await auth.api.signInMagicLink({
-        body: { email },
-        headers: c.req.raw.headers,
-        asResponse: true,
-      });
+    const { email, redirect_url } = body;
 
-      if (!result.ok) {
-        const error = await result.json();
-        console.log("signInMagicLink error:", error);
-        return c.redirect(
-          "/login/magic-link?error=" +
-            encodeURIComponent(error.message || "Failed to send magic link"),
-        );
-      }
+    const intermediateURL = `/login/magic-link?success=${encodeURIComponent("Magic link sent! Check your email.")}`;
 
-      console.log("Magic link sent successfully");
-      return c.redirect(
-        "/login/magic-link?success=" +
-          encodeURIComponent("Magic link sent! Check your email."),
-      );
-    } catch (error) {
-      console.error("signInMagicLink catch error:", error);
-      return c.redirect(
-        "/login/magic-link?error=" + encodeURIComponent("Network error"),
-      );
-    }
-  })
-  .get("/magic-link", async (c) => {
-    const token = c.req.query("token");
-
-    if (!token) {
-      return c.redirect(
-        "/login?error=" + encodeURIComponent("Invalid magic link"),
-      );
-    }
-
-    try {
-      const result = await auth.api.magicLinkVerify({
-        body: { token },
-        headers: c.req.raw.headers,
-        asResponse: true,
-      });
-
-      if (!result.ok) {
-        const error = await result.json();
-        return c.redirect(
-          "/login?error=" +
-            encodeURIComponent(
-              error.message || "Invalid or expired magic link",
-            ),
-        );
-      }
-
-      // Forward cookies from magic link verification
-      forwardCookies(result, c);
```

Comment:
This is gone because there are no cookies to bet set yet. And better-auth sets everything when people click the link.